pkg/apis/serving/v1alpha1/service_validation.go

```diff
@@ -180,3 +181,27 @@ func (rt *ReleaseType) Validate(ctx context.Context) *apis.FieldError {
 
 	return errs.Also(rt.Configuration.Validate(ctx).ViaField("configuration"))
 }
+
+func (s *Service) validateCreatorAndModifier(ctx context.Context) *apis.FieldError {
+	if len(s.GetAnnotations()) == 0 {
+		return nil
+	}
+	var errs *apis.FieldError
+	if !apis.IsInUpdate(ctx) {
+		return nil
+	}
+	original := apis.GetBaseline(ctx).(*Service)
+	if original.GetAnnotations()[serving.CreatorAnnotation] != s.GetAnnotations()[serving.CreatorAnnotation] {
+		errs = errs.Also(&apis.FieldError{
+			Message: fmt.Sprintf("annotation %q is immutable", serving.CreatorAnnotation),
```

Comment:
I don't think you want annotation in the error, since the error will be `annotaion X is immutable: metadata.annotations.X`, I think you just want `annotation value is immutable` and the path will show which one.